lib/pbench/server/database/models/datasets.py

```diff
@@ -831,10 +849,14 @@ def getvalue(
             # the stored data; let the caller know with an exception.
             if type(value) is not dict:
                 raise MetadataBadStructure(dataset, key, name)
-            if i not in value:
+            if i in value:
+                name = i
+                value = value[i]
+            elif i.lower() in value:
+                name = i.lower()
+                value = value[name]
+            else:
                 return None
```

Comment:
Alternately,
```
            if i in value:
                name = i
            elif i.lower() in value:
                name = i.lower()
            else:
                return None
            value = value[name]
```
This fetches the value only once, once we've figured out its name.

Alternately alternately,
```
            name = i
            v = value.get(name)
            if not v:
                name = i.lower()
                v = value.get(name)
                if not v:
                    return None
            value = v
```
This avoids the redundant key searches:  get the value for the name when we check to see if the key is in the dictionary, and only search again if we don't find it, rather than doing two or three searches.

Reply:
> The second is wrong

Ah, yes...quite so.

> The first would work

What I didn't like about your proposed code was the dissimilarity between lines 854 and 857; and, once we make them match, then the result is itching to be pulled out into a common path.  🙂